setup/base.go

```diff
@@ -311,6 +312,17 @@ func (b *BaseDendrite) SetupAndServeHTTP(
 		}
 	}
 
+	externalRouter.Handle("/health",
```

Comment:
My feeling is that this almost certainly wants to be on `internalRouter`, not `externalRouter`. Otherwise it'll be exposed on the public mux in monolith mode, and it won't be in the right place with `-api` set.

Reply:
@neilalexander and I discussed this a bit in a sync call but it's not a simple problem because we have to contend with monolith and polylith mode and making it available in each mode.

We probably want this on `internalRouter` because in the future, it will share more information about the state of databases, etc (see https://github.com/matrix-org/dendrite/issues/1429)

But we might need to put metrics and health together and add some new logic to make it work.